openeo/rest/datacube.py

```diff
@@ -1900,6 +1900,86 @@ def validate(self) -> List[dict]:
     def tiled_viewing_service(self, type: str, **kwargs) -> Service:
         return self._connection.create_service(self.flat_graph(), type=type, **kwargs)
 
+    def _get_spatial_extent_from_load_collection(self):
+        pg = self.flat_graph()
+        for node in pg:
+            if pg[node]["process_id"] == "load_collection":
+                if "spatial_extent" in pg[node]["arguments"] and all(
+                    cd in pg[node]["arguments"]["spatial_extent"]
+                    for cd in ["east", "west", "south", "north"]
+                ):
+                    return pg[node]["arguments"]["spatial_extent"]
+        return None
+
+    def preview(
+        self,
+        center: Union[Iterable, None] = None,
+        zoom: Union[int, None] = None,
+    ):
+        """
+        Creates a service with the process graph and displays a map widget. Only supports XYZ.
+
+        :param center: (optional) Map center. Default is (0,0).
+        :param zoom: (optional) Zoom level of the map. Default is 1.
+
+        :return: ipyleaflet Map object and the displayed Service
+        """
+        if "XYZ" not in self.connection.list_service_types():
+            raise OpenEoClientException("Backend does not support service type 'XYZ'.")
+
+        # Check we are inside Jupyer Notebook
+        try:
+            shell = get_ipython().__class__.__name__
+            if shell == "ZMQInteractiveShell":
+                try:
+                    from IPython.display import display
+                    import ipyleaflet
+                except ImportError:
```

Comment:
I think it's cleaner to first do the `import IPython` attempt, before doing that magic `get_ipython`  call. The latter will only work anyway if the former worked.

I'd also  avoid calling this magic `get_ipython` function that is nowhere imported, which trips up linting tools 
If I understand correctly it can be imported explicitly with 
```python
import IPython.core.getipython
IPython.core.getipython.get_ipython()
```

Reply:
Should be fixed.